### ISSUE TITLE
About "upgrade_source_domain"

### ISSUE THREAD

**Issue by liujianzhao6328057** [Evgeneus/Graph-Domain-Adaptaion#3]
Hi, thanks for your inspiring work! I have read the code and have some doubt about it. During the curriculum learning, why not remove the pseudo labelled target samples from the target domain after they are added to the source domain?
Expect your reply. Thanks.

**Comment by Evgeneus** [comment]
Hi, thanks for your interest and question!

MTD framework assumes that there are several target domains given (with unlabeled images) you need to adapt on.  As for the evaluation protocol, all these target domains with all images are used to measure the model performance. Therefore, there is no need to remove target images from the target domains.

Note after the curriculum episode on the chosen target domain T is finished, we only need to select high confident pseudo labels from T. After this, domain T is used only for the evaluation.

**Comment by liujianzhao6328057** [comment]
In the framework of CGCT, there is only one combined domain. Suppose that we select high confident pseudo labels from T after one curriculum step. At the next  curriculum step, we again select pseudo samples from the same T. If we don't mask (remove) the target samples which have been selected at the previous curriculum step, some samples will be selected again and may have differnet pseudo labels.   This will confuse the network training?

**Comment by liujianzhao6328057** [comment]
Besides, are the training data and the testing data of target domains the same? Don't we split the training and the testing images?

**Comment by Evgeneus** [comment]
> Besides, are the training data and the testing data of target domains the same?

Yes, only DomainNet dataset has train/test split. Note during the adaptation ground truth labels of target domains cannot be used.

> In the framework of CGCT, there is only one combined domain. Suppose that we select high confident pseudo labels from T after one curriculum step. At the next curriculum step, we again select pseudo samples from the same T. If we don't mask (remove) the target samples which have been selected at the previous curriculum step, some samples will be selected again and may have differnet pseudo labels. This will confuse the network training?

The current code is for D-CGCT. We will release CGCT code soon. We never have image duplicates. 
Though, as the training evolves one can potentially revise pseudo labels every N iters.

**Comment by liujianzhao6328057** [comment]
Expect your code and appreciate your patient reply.

**Comment by Evgeneus** [comment]
Hi @liujianzhao6328057!  Updated the repo with CGCT code!
